### PR TITLE
13 onboarding flow backend

### PR DESCRIPTION
**Pull request by rkerins91**
Contains user/:id/experience PUT route to update user's experience, as well as the user schema to have type mixed on experience prop.

Updated route to make fewer DB calls and added pre-defined array of languages to .env file.

add following line to .env
`availableLanguages = ["C","C++","Java","JavaScript","Python","Ruby"]`

### CODE REVIEW

File: server/routes/user.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ router.post(
     }
     const { name, email } = req.body;
     try {
-      const user = new User({ name, email });
+      const user = new User({ name, email, experience: {} });
```
Comment (ashan36) [inline server/routes/user.js]: I think this might conflict with my PR a bit. I can easily update mine to include an empty experience object.

**Review by ashan36 — CHANGES_REQUESTED** [review]
I'll update my PR to add the experience object to the new user. If you just remove that part on yours then you can merge it into dev without any conflicts popping up when I merge mine.

**Comment by rkerins91** [comment]
> I'll update my PR to add the experience object to the new user. If you just remove that part on yours then you can merge it into dev without any conflicts popping up when I merge mine.

Alright, done.

**Review by ashan36 — APPROVED** [review]

File: server/routes/user.js [hunk 2]
```diff
@@ -28,4 +28,19 @@ router.post(
...
 );
 
+router.put("/:id/experience", async (req, res) => {
+  // Get language data from body of request, userID from params
+  const languages = req.body;
+  const user = await User.findById(req.params.id);
+  try {
+    // loop through body of request, setting user experience for each key
+    for (let language in languages) {
+      await user.experience.set(language, Number(languages[language]));
```
Comment (skassam21) [inline server/routes/user.js]: maybe we should validate languages against a predefined list

Also, is there a way to set it all at once as opposed to doing n calls to the database to set this information on the user? It's not that efficient

File: server/routes/user.js [hunk 3]
```diff
@@ -28,4 +28,36 @@ router.post(
   }
 );
 
+router.put("/:id/experience", async (req, res) => {
+  const languages = { ...req.body };
+  const user = await User.findById(req.params.id);
+  const availableLanguages = [
```
Comment (skassam21) [inline server/routes/user.js]: probably want this list to be as a constant somewhere else in case we need to reuse it

**Pull request merged**